lib/Service/RetentionService.php

```diff
@@ -152,6 +179,17 @@ public function executeRetentionPolicy(IUser $user): ?bool {
 			return true;
 		}
 
+		// Check if we disable the user
+		try {
+			$this->shouldPerformActionOnUser($user, $skipDisableIfNewerThan);
+
+			$user->setEnabled(false);
+			$this->logger->info('Account disabled: ' . $user->getUID());
+			return true;
```

Comment:
This will execute every day and basically prevent accounts from being deleted.

Maybe we can wrap it with a check if the user is disabled already:
```suggestion
			if ($user->isEnabled()) {
				$user->setEnabled(false);
				$this->logger->info('Account disabled: ' . $user->getUID());
				return true;
			}
			$this->logger->debug('Account already disabled, continuing with potential deletion: ' . $user->getUID());
```

But actually you still disable all users when `$skipDisableIfNewerThan === 0`
And also you delete all users when `$skipIfNewerThan === 0`

As the return above is linked with `&&`.

So reach try needs to be wrapped with `$skipDisableIfNewerThan !== 0` and   `$skipIfNewerThan !== 0`
respectively?